### PR TITLE
[Fluent 2 iOS] SegmentedControl colors update

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

`SegmentedControl` was updated to match fluent 2 specs.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_light_1 | after_light_1 |
| before_light_2 | after_light_2 |
| before_dark_1 | after_dark_1 |
| before_dark_2 | after_dark_2 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

### CODE REVIEW

**Comment by harrieshin** [comment]
disabled primary style segment control is that by design?

**Comment by laminesm** [comment]
> disabled primary style segment control is that by design?

Yes it's what's what's spec'd on the figma

File: ios/FluentUI/SegmentedControl/SegmentedControl.swift [hunk 1]
```diff
@@ -272,20 +245,27 @@ open class SegmentedControl: UIControl {
...
-        guard let window = window else {
-            return
-        }
+    public func updateColors() {
 
-        pillMaskedContentContainerView.backgroundColor = customSegmentedControlSelectedButtonBackgroundColor ?? (isEnabled ? style.selectionColor(for: window) : style.selectionColorDisabled)
-        backgroundView.backgroundColor = customSegmentedControlBackgroundColor ?? (isEnabled ? style.backgroundColor(for: window) : style.backgroundColorDisabled(for: window))
-        let maskedContentColor = isEnabled ? (customSelectedSegmentedControlButtonTextColor ?? style.segmentTextColorSelected(for: window)) : style.segmentTextColorSelectedAndDisabled(for: window)
+        pillMaskedContentContainerView.backgroundColor = customSegmentedControlSelectedButtonBackgroundColor ?? style.selectionColor(fluentTheme: fluentTheme)
+        backgroundView.backgroundColor = customSegmentedControlBackgroundColor ?? style.backgroundColor(fluentTheme: fluentTheme)
```
Comment (edjamesmsft) [inline ios/FluentUI/SegmentedControl/SegmentedControl.swift:267]: Is there a reason you got rid of the disabled background color? Was it removed from design?

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/SegmentedControl/SegmentedControl.swift (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI/SegmentedControl/SegmentedControl.swift:267]: It's identical to background color so there's no need to have a separate function for it

**Pull request merged**